src/Adapter/Product/SpecificPrice/Update/SpecificPricePriorityUpdater.php

```
@@ -52,33 +83,56 @@ public function setPrioritiesForProduct(ProductId $productId, PriorityList $prio
     {
         try {
             if (!SpecificPrice::setSpecificPriority($productId->getValue(), $priorityList->getPriorities())) {
-                throw new CannotSetSpecificPricePrioritiesException(sprintf(
-                    'Failed to set specific price priorities for product #%d',
-                    $productId->getValue()
-                ));
+                throw $this->buildExceptionForProductPriorityError($productId);
```

This one should trigger a `CannotUpdateSpecificPriceException` it's an identified error use case

---

I still think `CannotUpdateSpecificPriceException` is redundant. 
As I mentioned before, there is only one scenario when it should throw a domain exception - when provided priorites are invalid - and it's already covered by SpecificPriceConstraintException thrown in PriorityList VO. In all other cases (including this one) it fails due to infrastructural issues so it should be CoreException.

---

sorry we were probably having in mind `CannotSetSpecificPricePrioritiesException`, that one is not used anymore, so it should be deleted.